test/functional/provider/perl_spec.lua

```diff
@@ -19,7 +23,51 @@ before_each(function()
   clear()
```

Comment:
seems like the `missing_provider()` check (line 15) is not working, or the `provider#perl#Detect()` function is not working, since the tests are failing on the BSD CIs : 

```
[ RUN      ] legacy perl provider :perl command: 32.24 ms ERR
test/functional/helpers.lua:106: Vim(echoerr):"Neovim::Ext" cpan module is not installed

stack traceback:
	test/functional/helpers.lua:106: in function 'request'
	test/functional/helpers.lua:270: in function 'command'
	test/functional/provider/perl_spec.lua:34: in function <test/functional/provider/perl_spec.lua:33>

[ RUN      ] legacy perl provider :perlfile command: 34.05 ms ERR
test/functional/helpers.lua:106: Vim(echoerr):"Neovim::Ext" cpan module is not installed

stack traceback:
	test/functional/helpers.lua:106: in function 'request'
	test/functional/helpers.lua:270: in function 'command'
	test/functional/provider/perl_spec.lua:41: in function <test/functional/provider/perl_spec.lua:38>

Reply:
i think this patch is needed : 

```diff
diff --git a/test/functional/helpers.lua b/test/functional/helpers.lua
index e8435cd3b796..5864cecb1377 100644
--- a/test/functional/helpers.lua
+++ b/test/functional/helpers.lua
@@ -768,10 +768,10 @@ function module.new_pipename()
 end
 
 function module.missing_provider(provider)
-  if provider == 'ruby' or provider == 'node' or provider == 'perl' then
+  if provider == 'ruby' or provider == 'node' then
     local prog = module.funcs['provider#' .. provider .. '#Detect']()
     return prog == '' and (provider .. ' not detected') or false
-  elseif provider == 'python' or provider == 'python3' then
+  elseif provider == 'python' or provider == 'python3' or provider == 'perl' then
     local py_major_version = (provider == 'python3' and 3 or 2)
     local errors = module.funcs['provider#pythonx#Detect'](py_major_version)[2]
     return errors ~= '' and errors or false